tests/test_remote.py

```diff
@@ -41,3 +46,106 @@ def test_response(self):
             self.assertEqual(mock.lines, [bytes2fsn(b"foo", None)])
             with open(fn, "rb") as h:
                 self.assertEqual(h.read(), b"resp")
+
+
+@skipIf(is_windows(), "unix only")
+class TUnixRemoteFifoFullCycle(TestCase):
+    @pytest.fixture(autouse=True)
+    def tmp_fifo_path(self, tmp_path):
+        self.registry = Mock(resp=bytes2fsn(b"response", None))
+        self.tmp_path = tmp_path
+        with mock.patch.object(QuodLibetUnixRemote, "_PATH", str(tmp_path / "control")):
+            yield
+
+    def _send_message_remote_proc(self, msg, callback):
+        """Execute QuodLibetUnixRemote.send_message() in a child process"""
+        # Using a child process is the only way to execute send_message, as it
+        # can only be run on the main thread, and blocks waiting for the
+        # response. It can't be run in a separate thread, because it uses
+        # signals to handle timeouts and we can't run the GLib mainloop in a
+        # thread as Gtk has already been initialised on the Python main thread.
+        # Attempting to use the loop in a thread leads to segfaults.
```

Comment:
I tried the thread approach first, but that segfaults in `gdk_event_prepare()` on macos. That's not that surprising given the explicit warnings against this in the [GDK threading documentation](https://developer.gnome.org/gdk3/stable/gdk3-Threads.html):

> GTK+, however, is not thread safe. You should only use GTK+ and GDK from the thread `gtk_init()` and `gtk_main()` were called on. This is usually referred to as the “main thread”.

Before the test is even run, `quodlibet._init()` has already called `Gtk.init_args()`.